jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py

```diff
@@ -41,6 +44,50 @@ def test_plugin(specviz_helper, spectrum1d):
             assert len(result_dict.get('uncertainty')) > 0
 
 
+@pytest.mark.filterwarnings('ignore:No observer defined on WCS')
+def test_spatial_subset(cubeviz_helper):
+    """
+    Tests that the spatial selection returns any valid result, DOES NOT VALIDATE THE VALUE
+    Not checking the value attempts to circumvent the issue we ran into here:
+    https://github.com/spacetelescope/jdaviz/pull/1564#discussion_r949427663
+    """
+
+    flux = np.arange(250).reshape((5, 5, 10)) * u.Jy
+    wcs_dict = {"CTYPE1": "RA---TAN", "CTYPE2": "DEC--TAN", "CTYPE3": "WAVE-LOG",
+                "CRVAL1": 205, "CRVAL2": 27, "CRVAL3": 4.622e-7,
+                "CDELT1": -0.0001, "CDELT2": 0.0001, "CDELT3": 8e-11,
+                "CRPIX1": 0, "CRPIX2": 0, "CRPIX3": 0}
+    w = WCS(wcs_dict)
+    large_cube = Spectrum1D(flux=flux, wcs=w)
+
+    label = "Test Cube"
+    cubeviz_helper.load_data(large_cube, data_label=label)
+
+    # add a region and rerun stats for that region
+    cubeviz_helper.app.get_viewer('flux-viewer').apply_roi(RectangularROI(-0.5, 0.5, 0.5, 2.5))
+    cubeviz_helper.app.get_viewer('flux-viewer').session.edit_subset_mode._mode = NewMode
+    # The cube only has one slice, so we have to grab the entire viewer's range to create a region
+    cubeviz_helper.app.get_viewer('spectrum-viewer').apply_roi(
+        XRangeROI(cubeviz_helper.app.get_viewer('spectrum-viewer').state.x_min,
+                  cubeviz_helper.app.get_viewer('spectrum-viewer').state.x_max))
+
+    cubeviz_helper.app.state.drawer = True
+
+    plugin = cubeviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    plugin.open_in_tray()
```

Comment:
Ok I've hammered this PR as far as I could, and now I have to concede I'm stuck and have no idea why the plugin is refusing to give this test its values

So currently, the tests are failing because the plugin results are still nans at the check. I can't determine why this is; when I copy and paste this entire test into a jupyter notebook, it returns the results as expected.

Slightly related is the only way I how to actually trigger the recalculation of the results via the API is by opening the tray, as the other tests in this suite do as well. As I understand it, this line is supposed to force the plugin to calculate the results, but it's not working. I've also tried calling `plugin._calculate_statistic` manually, but doesn't seem to work; it returns nans if the tray isn't opened. If the tray is open, it doesn't change the behavior.

The only thing I can think of is that there's a race condition between when the fields are being set and when the results are being read. But even that I find unlikely considering the results are supposed to be calculated for the default values immediately when the tray is opened.

Hopefully someone can point out where the mistake I'm making is. Please feel free to commandeer this PR if you find it if I can't get to it!

Reply:
If you can reproduce this locally, I would suggest you copy paste the test case into a notebook and debug there, if you have not already. If you can reproduce this failure that way, you can then drill down on the individual elements and so on.

Or is this the one you said you cannot reproduce locally and only on CI? I cannot keep track of all the conversations.

Reply:
> is this the one you said you cannot reproduce locally and only on CI?

Yep, this is the one... My local pytest runs give me the green light. @kecnry graciously tested it and saw the same behavior as I. I suppose the next step as Kyle suggested in debugging is adding print statements and seeing what the logs show?

Reply:
Did you try to use tox and run the exact tox env setting?

Reply:
I haven't gone quite that far since it takes a while to run, but perhaps it is warranted here... let me give that a go and let it run in the background

Reply:
You said on JIRA you were able to reproduce this failure finally, right?

Reply:
Oops, was that what was implied! Not at all, quite the opposite. Still trying to reproduce. When I ran the tox command, I couldn't get the error to fire either

Reply:
Actually, @kecnry when you tried running it, did you run it directly via pytest or tox?

Reply:
I never use tox if I can help it.

Reply:
Hmm. I guess I'll have a look and see...

Reply:
#1627 somehow has the failing job here passing but the rest fails with that same error... 🤯

Reply:
Update: Hopefully CI will pass now.

Reply:
Green checks! 🎉